src/mainnet/concentrators/balancer/AuraBalConcentrator.sol

```diff
@@ -66,60 +67,18 @@ contract AuraBalConcentrator is BalancerOperations, AMMConcentratorBase {
             IERC20(AURABAL).safeApprove(_compounder, type(uint256).max);
         }
 
-    /********************************** Mutated Functions **********************************/
-
-    /// @notice See {AMMConcentratorBase - depositSingleUnderlying}
-    function depositSingleUnderlying(uint256 _underlyingAmount, address _underlyingAsset, address _receiver, uint256 _minAmount) external payable override nonReentrant returns (uint256 _shares) {
-        if (!_isUnderlyingAsset(_underlyingAsset)) revert NotUnderlyingAsset();
-        if (!(_underlyingAmount > 0)) revert ZeroAmount();
-
-        _updateRewards(msg.sender);
-
-        if (msg.value > 0) {
-            if (msg.value != _underlyingAmount) revert InvalidAmount();
-            if (_underlyingAsset != ETH) revert InvalidAsset();
-        } else {
-            IERC20(_underlyingAsset).safeTransferFrom(msg.sender, address(this), _underlyingAmount);
-        }
+    /********************************** Internal Functions **********************************/
 
-        uint256 _assets = _addLiquidity(address(asset), _underlyingAsset, _underlyingAmount);
+    function _swapFromUnderlying(address _underlyingAsset, uint256 _underlyingAmount, uint256 _minAmount) internal override returns (uint256 _assets) {
+        _assets = _addLiquidity(address(asset), _underlyingAsset, _underlyingAmount);
         if (!(_assets >= _minAmount)) revert InsufficientAmountOut();
-
-        _shares = previewDeposit(_assets);
-        _deposit(msg.sender, _receiver, _assets, _shares);
-
-        _depositStrategy(_assets, false);
-
-        return _shares;
     }
 
-    /// @notice See {AMMConcentratorBase - redeemSingleUnderlying}
-    function redeemSingleUnderlying(uint256 _shares, address _underlyingAsset, address _receiver, address _owner, uint256 _minAmount) public override nonReentrant returns (uint256 _underlyingAmount) {
-        if (!_isUnderlyingAsset(_underlyingAsset)) revert NotUnderlyingAsset();
-        if (_shares > maxRedeem(_owner)) revert InsufficientBalance();
-        
-        _updateRewards(_owner);
-
-        uint256 _assets = previewRedeem(_shares);
-        _withdraw(msg.sender, _receiver, _owner, _assets, _shares);
-
-        _withdrawStrategy(_assets, _receiver, false);
-
+    function _swapToUnderlying(address _underlyingAsset, uint256 _assets, uint256 _minAmount) internal override returns (uint256 _underlyingAmount) {
         _underlyingAmount = _removeLiquidity(address(asset), _underlyingAsset, _assets);
         if (!(_underlyingAmount >= _minAmount)) revert InsufficientAmountOut();
```

Comment:
Is this method still supposed to swap to underlying? It looks like it only removes liquidity anymore without the swapping.

Reply:
I guess that `swap` is more of a terminology thing - the idea is to convert (swap/exchange/etc) the vault's primary asset into underlying assets, and vice versa.

Reply:
I guess that `swap` is more of a terminology thing - the idea is to convert (swap/exchange/etc) the vault's primary asset into underlying assets, and vice versa.